fineract-provider/src/main/java/org/apache/fineract/portfolio/loanproduct/serialization/LoanProductDataValidator.java

```diff
@@ -809,6 +809,14 @@ public void validateForCreate(final JsonCommand command) {
                     "Progressive repayment schedule processing is only available with `Advanced payment allocation` strategy");
         }
 
+        if (LoanScheduleType.PROGRESSIVE.equals(LoanScheduleType.valueOf(loanScheduleType))
```

Comment:
Please extract this validation into a method and that way can be reused when loan got submitted or when loan got modified!

Reply:
Done! code updated!